tests/pylint/runpylint.py

```diff
@@ -68,18 +68,26 @@ def _get_py_paths(self, directory):
                 continue
 
             for f in files:
+                file_path = root + "/" + f
                 try:
-                    with open(root + "/" + f) as fo:
+                    if os.path.getsize(file_path) > 10e6:
+                        # Skip too large files - 10 MB should be enough for everyone!
+                        continue
+                    with open(file_path) as fo:
                         lines = fo.readlines()
                 except UnicodeDecodeError:
                     # If we couldn't open this file, just skip it.  It wasn't
                     # going to be valid python anyway.
                     continue
+                except FileNotFoundError:
+                    # This can happen with ui/webui/bots/images/* that is a symlink to nowhere.
+                    # These files are not Python scripts, so let's not even try.
```

Comment:
Is this not respected by the ignore-paths?

Reply:
No, we first scan here in this function, then pass that list to pylint, which filters by its own ignores. Yes, that's super ugly. Why? I don't know, it's already there.

Reply:
Doh, thanks!